### PR TITLE
use process id when thread id is not available

### PR DESCRIPTION
**Pull request by arnetheduck**
this works around #4, in a sense..

### CODE REVIEW

**Comment by stefantalpalaru** [comment]
In  "tests/other/threads.test" you should remove the `timestamp_peg=...` line, since you're ignoring that pid value for all tests.

Actually, that test and the associated "topics_and_loglvls_threads.nim" can now be removed completely. Instead of it, enable `--threads:on` for all tests.

**Comment by arnetheduck** [comment]
would probably make more sense to run tests once with threads and once without, but that's out of scope for this PR I think

**Comment by stefantalpalaru** [comment]
> And actually, "topics_and_loglvls_threads.nim" and "topics_and_loglvls_threads.nim.cfg" was never really needed in the first place

And how would you have dealt with random thread IDs present in the output that needs to be compared with a hardcoded version? Of course it was needed, before the thread/pid field got ignored for all tests.

**Comment by kdeme** [comment]
But that was taken care of with the (ab)use of the `timestamp_peg` in the threads.test file no? Or am I missing something here?

File: chronicles.nim [hunk 1]
```diff
@@ -136,6 +136,24 @@ else:
...
 
+when compileOption("threads"):
+  # With threads turned on, we give the thread id
+  # TODO: Does this make sense on all platforms? On linux, conveniently, the
+  #       process id is the thread id of the `main` thread..
+  proc getLogThreadId*(): int = getThreadId()
+else:
+  # When there are no threads, we show the process id instead, allowing easy
+  # correlation on multiprocess systems
+  when defined(linux) or defined(macosx):
```
Comment (yglukhov) [inline chronicles.nim]: This can be just `defined(posix)` to allow bsd and others into the party.

**Comment by stefantalpalaru** [comment]
> But that was taken care of with the (ab)use of the `timestamp_peg` in the threads.test file no?

Oh, you mean passing "threads:on" in the "*.test" file instead of a "*.nim.cfg". Got it.

File: chronicles.nim (hunk 1 above)
Comment (arnetheduck) [reply chronicles.nim]: fixed, nice

**Pull request merged**